### PR TITLE
Feat: Add a Favicon and Custom Title to pages

### PR DESCRIPTION
**Pull request by jmullercuber**
# What:
- Adding a favicon (pink dragon logo)
- All pages export the SEO component for Head (containing custom title and description)
- Add some settings to allow the site to open as a Progressive Web App (PWA): not particularly tested

# Why:
- To help users identify the BlazerBots webpage out of all their open browser tabs
- So the webpage appears better in search engine results
- To provide smooth user experience

# Testing:
- Home page: image
- "About Us" page: image
- The progressive web app config "tested" by inspecting Application in Firefox Dev Tools: image

### CODE REVIEW

File: gatsby-config.ts [hunk 1]
```diff
@@ -2,14 +2,28 @@ import type { GatsbyConfig } from "gatsby";
...
+    {
+      resolve: `gatsby-plugin-manifest`,
+      options: {
+        name: "Overland BlazerBots FRC Team 3807",
+        short_name: "BlazerBots",
+        start_url: "/",
+        background_color: "#ff007f",
+        theme_color: "#ff007f",
+        display: "standalone",
+        icon: "src/images/logo.svg", // This path is relative to the root of the site.
```
Comment (jmullercuber) [inline gatsby-config.ts:23]: This is reusing the logo we already have defined in `images/` for the favicon (`gatsby-plugin-manifest` takes care of resizing). I posted a question in Teams regarding using alternate versions. If we decide to use a different icon/color scheme all we will need to do is upload that version, and point to that file here.

File: src/pages/history.tsx [hunk 2]
```diff
@@ -12,20 +15,24 @@ const carouselItemStyle = {
   width: "100%"
 }
 
-const makeCarouselItem = (image: string) => (
-  <Carousel.Item>
+const makeCarouselItem = (image: string, index: number) => (
+  <Carousel.Item key={index}>
```
Comment (jmullercuber) [inline src/pages/history.tsx:19]: This change is not related to the favicon/title work. It fixes a warning that appears in development (`Each child in a list should have a unique "key" prop`)

File: gatsby-config.ts (hunk 1 above)
Comment (samgoldman) [reply gatsby-config.ts:23]: As-is which version from teams is used?

**Review by aselenke — APPROVED** [review]

File: gatsby-config.ts (hunk 1 above)
Comment (jmullercuber) [reply gatsby-config.ts:23]: This would be option No. 6 (pink fill with transparent background). For the post in Teams I modified the SVG in various combinations to show how it could look (work to display those modifications are not part of this PR).
